src/nile/common.py

```diff
@@ -39,11 +43,7 @@ def get_gateways():
 
     except FileNotFoundError:
         with open(NODE_FILENAME, "w") as f:
-            networks = {
```

Comment:
In the try block above, gateway should be in plural.

Also, I think this is still not working for sending transactions if goerli2 is not in the node.json file, but the file exists. OTHER_NETWORKS are added just when the file is not found. Maybe return it in the try block as well?

Reply:
If we do this, I think it makes sense to remove writing the DEFAULT_NETWORKS to the node.json file at all, because it will be part of GATEWAYS always anyway.

Reply:
Good idea!

Reply:
> I think it makes sense to remove writing the DEFAULT_NETWORKS to the node.json file

It seems difficult to handle (at least, to cleanly handle) `localhost` network calls when the node.json is not present with this approach^. I think we'd have to add conditionals to `get_gateway_url` and `get_feeder_url` in starknet_cli.py. Maybe there's a better way. If it's agreeable, I'd say let's make another issue to track that for the future.

Reply:
Why do we need to handle localhost calls when the node.json is not present with this approach? The suggestion is to remove DEFAULT_NETWORKS, which are goerli2 and integration, which gateway_urls won't probably change often anyway. localhost calls would still be going through node.json. Sorry if I'm missing something.

Reply:
Forgive me, I did not provide enough context in my response.

The idea was to handle `localhost` separately from `DEFAULT_NETWORKS` through some sort of conditional. Otherwise, a node.json will be created (if it's not already) for txs querying the `DEFAULT_NETWORKS`. This doesn't break anything; however, it's not ideal. This occurs because `GATEWAYS.get()` returns None when a key does not exist instead of an error that we can handle.

I'm interested in the networks config idea.

Reply:
Got it. 

My issue with the current PR state is that it is not addressing the real issue IMO, which is the UX using Nile. We can use a different approach than the one I suggested in this thread (mine is probably not the best for sure), but letting this as it is, I fear users will upgrade Nile, and they will struggle with sending transactions to goerli2 or integration. You and I both know how to fix it, but I feel the error they will receive is not well documented (and I think we could prevent the issue by default). Is the details what makes great tooling IMO (hardhat vs truffle).

Reply:
this is my middle ground suggestion:

- let's not write any node.json. it's confusing pollution to the user, and it's information nile already knows. no more writing conflicts nor retrocompatibility issues.
- let's still read node.json. if there's any user custom network, let's add it (or overwrite any default). this is the stem for an actual nile config file, but retrocompatible and it buys us time to design properly, while fixing this.

thougths?